unit/av_test.c

```diff
@@ -995,10 +995,11 @@ run_test_set()
 	if (bad_address != NULL) {
 		printf("Testing with bad_address = \"%s\"\n", bad_address);
 		failed += run_tests(test_array_bad, err_buf);
+	
+		bad_address = NULL;
+		printf("Testing with invalid address\n");
+		failed += run_tests(test_array_bad, err_buf);
 	}
-	bad_address = NULL;
-	printf("Testing with invalid address\n");
-	failed += run_tests(test_array_bad, err_buf);
```

Comment:
If I understand the change, the tests for invalid addresses is only done if the user inputs a set of addresses which will fail.